### PR TITLE
Rdi 53 zenodo transform

### PR DESCRIPTION
**Pull request by ehanson8**
#### Helpful background context
JPAL records don't require any parsing of identifiers for the `source_link` so I don't think a JPAL subclass is necessary at this point but let me know if you think differently.

#### How can a reviewer manually see the effects of these changes?
Apologies for continuing to stall this discussion, let's discuss when we reengage with the project 

#### What are the relevant tickets?
https://mitlibraries.atlassian.net/browse/RDI-53

#### Developer
- [x] All new ENV is documented in README
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer
- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

#### Includes new or updated dependencies?
NO

### CODE REVIEW

**Review by hakbailey — CHANGES_REQUESTED** [review]
One important change: we need to add `zenodo` to the cli `--source` choices, and update the cli logic to handle that source. Other non-blocking suggestions inline and below.

A general comment: we could simplify the fixtures by combining all of the `relatedItemIdentifier` types into a single fixture that just has multiple relatedItems. Same for the `nameIdentifie`r types. It looks like you've updated the `all_fields` fixture to include all the variations anyway, so we could even just use that and reduce the need for so many extra fixtures.

File: transmogrifier/sources/datacite.py [hunk 1]
```diff
@@ -201,6 +201,30 @@ def create_from_datacite_xml(
...
+            if "alternateIdentifierType" in alternate_identifier.attrs:
+                i.kind = alternate_identifier.attrs["alternateIdentifierType"]
+            kwargs["identifiers"].append(i)
+
+        related_identifiers = xml.metadata.find_all("relatedIdentifier")
+        for related_identifier in [
+            i
+            for i in related_identifiers
+            if "relationType" in i.attrs
+            and i.attrs["relationType"] in ["IsIdenticalTo", "IsVersionOf"]
```
[the remark this answers is not in the document]
Comment (ehanson8) [reply transmogrifier/sources/datacite.py:218]: I think we should remove it here, I agree about the potential for confusion and that `isIdenticalTo` should remain. I'll update the tests

**Comment by ehanson8** [comment]
Good call on the fixtures, I fleshed out the `all_fields` fixture and removed the redundant ones

**Review by hakbailey — APPROVED** [review]
One non-blocking cleanup suggestion inline, but this looks good!

**Pull request merged**